pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/segmentgenerationandpush/SegmentGenerationAndPushTaskGenerator.java

```diff
@@ -177,6 +180,78 @@ public String getTaskType() {
     return pinotTaskConfigs;
   }
 
+  @Override
+  public List<PinotTaskConfig> generateAdhocTasks(TableConfig tableConfig, Map<String, String> taskConfigs)
+      throws Exception {
+    String taskUUID = UUID.randomUUID().toString();
+    // Only generate tasks for OFFLINE tables
+    String offlineTableName = tableConfig.getTableName();
+    if (tableConfig.getTableType() != TableType.OFFLINE) {
+      LOGGER.warn("Skip generating SegmentGenerationAndPushTask for non-OFFLINE table: {}", offlineTableName);
+      return ImmutableList.of();
+    }
+
+    TableTaskConfig tableTaskConfig = tableConfig.getTaskConfig();
+    if (tableTaskConfig != null) {
+      Map<String, String> taskConfigsFromTable =
+          tableTaskConfig.getConfigsForTaskType(MinionConstants.SegmentGenerationAndPushTask.TASK_TYPE);
+      if (taskConfigsFromTable != null) {
+        taskConfigsFromTable.entrySet().forEach(entry -> taskConfigs.putIfAbsent(entry.getKey(), entry.getValue()));
```

Comment:
should user-provided task configs overwrite those from TableConfig instead? but anyway, perhaps leave a comment about merging the configs.

Reply:
my thought is to take the task config from the table as the base, then apply the changes from ad-hoc task config.